contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -592,54 +597,9 @@ else if (newWaitingPartition.size() != 0) {
           logger.info("[ONE_TO_MANY]: Create operator partition for kafka partition(s): {} ", StringUtils.join(kps[i], ", "));
           newPartitions.add(createPartition(kps[i], initOffset, newManagers));
         }
-
-      }
-      else if (newWaitingPartition.size() != 0) {
-
-        logger.info("[ONE_TO_MANY]: Add operator partition for kafka partition(s): {} ", StringUtils.join(newWaitingPartition, ", "));
-        partitions.add(createPartition(Sets.newHashSet(newWaitingPartition), null, newManagers));
+        newWaitingPartition.clear();
         windowDataManager.partitioned(newManagers, deletedOperators);
```

Comment:
Not sure if it is a existing bug but windowDataManager is always called, right? Actually if we don't delete partition, we don't have to call this method?